src/shell/services/instance.ts

```diff
@@ -99,6 +99,23 @@ export const instanceApi = createApi({
       // Restore cache once content/schema uses rtk query for mutations and can invalidate this
       keepUnusedDataFor: 0.0001,
     }),
+    searchContent: builder.query<ContentItem[], SearchQuery>({
+      query: (options) => {
+        const params = new URLSearchParams();
+        params.append("q", options.query);
+        // UnorderedQuery
+        if (options.limit) params.append("limit", options.limit.toString());
+        if (options.startDate) params.append("startDate", options.startDate);
+        if (options.endDate) params.append("endDate", options.endDate);
+        // OrderedQuery
+        if ("order" in options) params.append("order", options.order);
```

Comment:
I think `options.order` would be more idiomatic in typical JS code, but TypeScript will error on compilation if you try to access `order` because it might not exist on type `Query` (because it does not exist on `UnorderedQuery`)

This is a moot point though, because I will just use an object to construct URLSearchParams like you've suggested.